src/api/java/appeng/api/features/IInscriberRegistry.java

```diff
@@ -0,0 +1,52 @@
+package appeng.api.features;
+
+
+import java.util.List;
+import java.util.Set;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import net.minecraft.item.ItemStack;
+
+
+/**
+ * Lets you manipulate Inscriber Recipes, by adding or editing existing ones.
+ *
+ * @author thatsIch
+ * @version rv2
+ * @since rv2
+ */
+public interface IInscriberRegistry
+{
+	/**
+	 * Current list of registered recipes, you can modify this if you want too.
+	 *
+	 * @return currentlyRegisteredRecipes
+	 */
+	@Nonnull
+	List<IInscriberRecipe> getRecipes();
+
+	/**
+	 * Optional items which are used in the top or bottom slot
+	 *
+	 * @return set of all optional items
+	 */
+	@Nonnull
+	Set<ItemStack> getOptionals();
+
+	/**
+	 * Get all registered items which are valid inputs
+	 *
+	 * @return set of all input items
+	 */
+	@Nonnull
+	Set<ItemStack> getInputs();
+
+	/**
+	 * add a new recipe the easy way, duplicates will not be added.
+	 * Added recipes will be automatically added to the optionals and inputs.
+	 *
+	 * @param recipe new recipe
+	 */
+	void addRecipe( @Nullable IInscriberRecipe recipe );
```

Comment:
I do not think this should be `@Nullable`, adding null as recipe is completely pointless.
As the annotations are mostly hints for the IDE, throwing an exception like `IllegalArgumentException` in case of null should be considered.